extensions/amp-access-laterpay/0.1/amp-access-laterpay.js

```diff
@@ -30,18 +35,120 @@ export class LaterpayVendor {
   constructor(accessService) {
     /** @private @const */
     this.accessService_ = accessService;
-    // TODO: implement
+    this.win_ = this.accessService_.win;
+    this.laterpayConfig_ = this.accessService_.adapter_.getConfig();
+    this.purchaseConfig_ = null;
+    this.purchaseOptionListeners_ = [];
+    this.selectedPurchaseOption_ = null;
+    this.purchaseButton_ = null;
+
+    /** @const @private {!Xhr} */
+    this.xhr_ = xhrFor(this.win_);
   }
 
   /**
    * @return {!Promise<!JSONType>}
    */
   authorize() {
-    const win = this.accessService_.win;
-    user().assert(isExperimentOn(win, 'amp-access-laterpay'),
+    user().assert(isExperimentOn(this.win_, 'amp-access-laterpay'),
         'Enable "amp-access-laterpay" experiment');
-    // TODO: implement
-    return Promise.resolve({access: true});
+    return this.getInitialPurchaseConfig_().then(purchaseConfig => {
+      this.purchaseConfig_ = purchaseConfig;
+      if (!purchaseConfig.access) {
+        this.renderPurchaseOverlay_();
+      }
+      return {access: purchaseConfig.access};
+    });
+  }
+
+  getInitialPurchaseConfig_() {
+    const urlPromise = this.accessService_.buildUrl_(CONFIG_URL,
+        /* useAuthData */ false);
+    return urlPromise.then(url => {
+      dev().fine(TAG, 'Authorization URL: ', url);
+      return this.accessService_.timer_.timeoutPromise(
+          AUTHORIZATION_TIMEOUT,
+          this.xhr_.fetchJson(url, {
+            credentials: 'include',
+            requireAmpResponseSourceOrigin: true,
+          }));
+    });
+  }
+
+  renderPurchaseOverlay_() {
+    const laterpayList = this.win_.document.querySelectorAll(
+      'amp-access-laterpay-list')[0];
+    const listContainer = document.createElement('ul');
+    // TODO set these up somewhere else and make them configurable
+    this.purchaseConfig_.premiumcontent.title = 'Buy this article';
```

Comment:
The possibility I had thought about was to have it be supplied via the in page configuration parameters, which means the merchant could just server render it for whatever language they wanted. This is similar to what we do with our existing integration and meta tags: https://connectormwi.laterpay.net/docs/inpage_configuration/translations.html